core/src/test/scala/shapeless/serialization.scala

```diff
@@ -1065,7 +1065,9 @@ class SerializationTests {
 
   @Test
   def testFunctor {
-    assertSerializable(Functor[Some])
+    val f = Functor[Some]
+    f.map(Some(2))(_.toString)
+    assertSerializable(f)
     assertSerializable(Functor[Option])
```

Comment:
A safety check, I changed it a bit along with the workarounds for 2.12. I don't remember if it still has to be here, or if it's independent of the other devs of this PR.

The problem is that the `Functor[Some]` instance found implicitly has some lazily initialized fields, in particular some `Generic1` / `IsHCons1` / `IsCCons1` (the first commit of this PR makes them serializable). Thus, serializability can change once these are made concrete. The call to `map` makes these fields concrete, and tests that.